src/Requests.jl

```diff
@@ -153,6 +156,11 @@
         r.finished = true
         close(p.sock)
 
+        #If MIME type allows, convert r.data to a Julia string
+        mimetype = r.headers["Content-Type"]
+        if (contains(mimetype, "charset=utf-8") || contains(mimetype, "charset=us-ascii"))
```

Comment:
But indeed there is something too magical about determining the return type at runtime from within the web stack. I'd much rather return the data always in raw bytes, and have user code call `bytestring` themselves on the data. But that would be a big breaking change for all existing code out there.